mesonbuild/coredata.py

```diff
@@ -418,24 +420,50 @@ def set_buildtype_from_others(self):
             mode = 'custom'
         self.builtins['buildtype'].set_value(mode)
 
-    def get_all_compiler_options(self):
-        # TODO think about cross and command-line interface. (Only .build is mentioned here.)
-        yield self.compiler_options.build
+    _V = TypeVar('_V')
 
-    def _get_all_nonbuiltin_options(self):
-        yield self.backend_options
-        yield self.user_options
-        yield from self.get_all_compiler_options()
-        yield self.base_options
+    @staticmethod
+    def get_prefixed_options_per_machine(
+        options_per_machine # : PerMachine[Dict[str, _V]]]
+    ) -> Dict[str, _V]:
+        return {
+            prefix + k: v
+            for for_machine, prefix in machines_and_prefixes
+            for k, v in options_per_machine[for_machine].items()
+        }
+
+    @staticmethod
+    def get_unprefixed_options_per_machine_keyset(
+        options_per_machine # : PerMachine[Dict[str, _V]]
+    ) -> FrozenSet[str]:
+        return frozenset(options_per_machine.build) | \
+                frozenset(options_per_machine.host) | \
```

Comment:
__[Flake8]__ [E126] continuation line over-indented for hanging indent ([link](https://sider.review/gh/repos/19784232/pulls/5263#issue-3979933))

<sub>posted by Sider</sub>